test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -1367,6 +1367,24 @@ Describe "Type inference Tests" -tags "CI" {
         $res.Name -join ' ' | Should -Be "System.IO.FileInfo System.IO.DirectoryInfo"
     }
 
+    It 'Infers closest variable type' {
+        $res = [AstTypeInference]::InferTypeOf( { [string]$TestVar = "";[hashtable]$TestVar = @{};$TestVar }.Ast)
```

Comment:
should there be some sort of order checking here?

Reply:
No. There shouldn't even be multiple inferred types but because assignments are included when inferring the output from a scriptblock I need that `Select -Last 1` statement.  
I tried fixing the issue earlier but it's tricky because sometimes assignments actually do output something.